cogs/games.py

```diff
@@ -20,18 +20,203 @@
+            if not (curr_msg.author in timeout_dict and
+                    (time() - timeout_dict[curr_msg.author]) < 3.0):
+                if curr_guess in "abcdefghijklmnopqrstuvwxyz" and len(
+                        curr_guess) == 1:
+                    await curr_msg.delete()
+                    if curr_guess in not_guessed:    # curr_guess in not_guessed => curr_guess is correct and new
+                        not_guessed.remove(curr_guess)
+                        first_line = "".join(
+                            char + " " if char not in not_guessed else "_ "
+                            for char in word).rstrip()
+                        player_msg_list.append(
+                            f"{curr_msg.author} guessed '{curr_guess}' correctly!"
+                        )
+                        if len(player_msg_list) > 3:
+                            player_msg_list = player_msg_list[-3:]
+                        txt_embed.set_field_at(
+                            0,
+                            name=f"hangman (category: {cat_name})",
+                            value=
+                            f"`{first_line}`\n```{HANG_LIST[num_mistakes]}```\n{NEWLINE.join(player_msg_list)}"
+                        )
+                        await hg_msg.edit(embed=txt_embed)
+                        if len(not_guessed) == 0:
+                            await ctx.send(
+                                f"congratulations {curr_msg.author}, you solved the hangman"
+                            )
+                    elif curr_guess in word:    # curr_guess not in not_guessed (elif) and in word => curr_guess is correct but already made
+                        player_msg_list.append(
+                            f"{curr_msg.author}, '{curr_guess}' was already guessed"
+                        )
+                        if len(player_msg_list) > 3:
+                            player_msg_list = player_msg_list[-3:]
+                        txt_embed.set_field_at(
+                            0,
+                            name=f"hangman (category: {cat_name})",
+                            value=
+                            f"`{first_line}`\n```{HANG_LIST[num_mistakes]}```\n{NEWLINE.join(player_msg_list)}"
+                        )
+                        await hg_msg.edit(embed=txt_embed)
+                    elif curr_guess not in incorrect_guesses:    # curr_guess not in not_guessed and not in word (elif) and not in incorrect guesses => curr_guess is incorrect and new
+                        num_mistakes += 1
+                        incorrect_guesses.add(curr_guess)
+                        last_line = f"incorrect guesses: {str(sorted(incorrect_guesses))[1:-1]}"
+                        player_msg_list.append(
+                            f"{curr_msg.author} guessed '{curr_guess}' wrong!")
+                        if len(player_msg_list) > 3:
+                            player_msg_list = player_msg_list[-3:]
```

Comment:
this code seems to be repeated a lot, so if it could be factored out maybe that would be useful?